### PR TITLE
Implement configurator class

### PR DESCRIPTION
**Pull request by bestwebua**
- [x] Implement gem configurator

```ruby
Truemail.configure do |config|
  # Required parameter. Should be an existing email on behalf of which
  # verification will be performed
  config.verifier_email = 'email@example.com'

  # Optional parameter. Should be an existing domain on behalf of which
  # verification will be performed.
  # By default verifier domain based on verifier email
  config.verifier_domain = 'somedomain.com'

  # Optional parameter. You can override default regex pattern
  config.email_pattern = /regex_pattern/
end
```

### CODE REVIEW

File: lib/truemail/error.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module Truemail
+  module Error
+    NOT_VALID_EMAIL = 'is invalid, use your real email address please'
```
Comment (alexxty7) [inline lib/truemail/error.rb]: Move to exceptions or exceptions to it

File: lib/truemail/configuration.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      @verifier_domain = email[REGEX_PATTERN, 3]
+    end
+
+    def verifier_domain=(domain)
+      raise ArgumentError.new(domain, 'valid domain') unless REGEX_DOMAIN_PATTERN.match?(domain.to_s)
+      @verifier_domain = domain
+    end
+
+    def complete?
+      !!verifier_email
```
Comment (alexxty7) [inline lib/truemail/configuration.rb]: present?

File: lib/truemail.rb [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+require 'truemail/exceptions'
+require 'truemail/error'
+require 'truemail/configuration'
 
 module Truemail
-  class Error < StandardError; end
+  class << self
+    attr_accessor :configuration
+
+    def configure
```
Comment (alexxty7) [inline lib/truemail.rb]: def configuration
 @configuration ||= begin do
     configuration = Configuration.new
     raise ConfigurationError, Error::CONFIGURATION_INCOMPLETE unless configuration.complete?
     configuration
 end
end

File: lib/truemail.rb [hunk 4]
```diff
@@ -1,5 +1,18 @@
...
+      return unless block_given?
+      self.configuration ||= Configuration.new
+      yield(configuration)
```
Comment (alexxty7) [inline lib/truemail.rb]: def configure
  yield configuration
end

File: lib/truemail/configuration.rb (hunk 2 above)
Comment (bestwebua) [reply lib/truemail/configuration.rb]: ```.present?``` isn't default ruby method ;)

File: lib/truemail.rb (hunk 4 above)
Comment (bestwebua) [reply lib/truemail.rb]: What about .configure without block?

File: lib/truemail.rb (hunk 3 above)
Comment (bestwebua) [reply lib/truemail.rb]: This code will never work. My refactor:

```ruby
module Truemail
  def self.configuration
    @configuration ||= begin
      return unless block_given?
      configuration = Configuration.new
      yield(configuration)
      raise ConfigurationError, ConfigurationError::INCOMPLETE_CONFIG unless configuration.complete?
      configuration
    end
  end

  def self.configure(&block)
    configuration(&block)
  end
end
```

File: lib/truemail/error.rb (hunk 1 above)
Comment (bestwebua) [reply lib/truemail/error.rb]: Fixed.

**Pull request merged**